src/main/java/es/usj/crypto/enigma/Machine.java

```diff
@@ -77,10 +77,12 @@ public String getCipheredText(String plainText) {
             // Plugboard substitution
             char output = plugboard.getPlug(input);
 
-            // Rotors position update
-            rightRotor.update(leftRotor, middleRotor);
-            middleRotor.update(leftRotor);
-            leftRotor.update();
+            // Rotors position update when a letter is encrypted
+            if (input >= 'A' && input <= 'Z') {
```

Comment:
May we use something like ALPHABET.charAt(0) instead of 'A' and the same for 'Z'?

Reply:
I didn't notice there was already an ordered alphabet!